packages/browser/src/integrations/offline.ts

```diff
@@ -0,0 +1,116 @@
+import { addGlobalEventProcessor, getCurrentHub } from '@sentry/core';
+import { Event, Integration } from '@sentry/types';
+import localforage from 'localforage';
+
+import { captureEvent } from '../index';
+
+/**
+ * store errors occuring offline and send them when online again
+ */
+export class Offline implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public readonly name: string = Offline.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Offline';
+
+  /**
+   * the key to store the offline event queue
+   */
+  private readonly _storrageKey: string = 'offlineEventStore';
+
+  public offlineEventStore: LocalForage;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(): void {
+    addGlobalEventProcessor(async (event: Event) => {
+      const self = getCurrentHub().getIntegration(Offline);
+      if (self) {
+        if (navigator.onLine) {
```

Comment:
This will break on environments that may miss this API (eg. some testing environments, or some odd devices (we had those in the paste)).
Can you use `getGlobalObject` from our `@sentry/utils` and check if `navigator` and `onLine` are there first? eg. `'onLine' in navigator`.